src/App.css

```diff
@@ -69,3 +75,24 @@
   border-radius: 50%;
   margin-right: 5px; /* Spacing between circles */
 }
+
+.square-filled {
+  display: flex;
+  width: 12px;
+  height: 12px;
+  background-color: #7bd310; /* Filled square color */
+  border: 1px solid #7bd310; /* Border color to have the same size as the empty version */
```

Comment:
It might be interesting to test this with black colour. Would the visualisation resemble a workflow a bit better?